addons/website_profile/controllers/main.py

```diff
@@ -225,12 +228,22 @@ def view_all_users_page(self, page=1, **searches):
             user_values = self._prepare_all_users_values(users)
 
             # Get karma position for users (only website_published)
-            position_domain = [('karma', '>', 1), ('website_published', '=', True)]
-            position_map = self._get_users_karma_position(position_domain, users.ids)
-            for user in user_values:
-                user['position'] = position_map.get(user['id'], 0)
+            if group_by:
+                position_map = self._get_users_karma_gain_position(group_by)
+                for user in user_values:
+                    u = position_map.get(user['id'], 0)
+                    user['position'] = 0 if u == 0 else u.get('position')
+                    user['karma_gain'] = 0 if u == 0 else u.get('karma_gain')
+                user_values = list(filter(lambda i: i['position'] != 0, sorted(user_values, key=lambda i: i['position'])))
+            else:
```

Comment:
if group_by from kwargs is not set, group_by will be 'all'.
So this will never bit hit, as group_by variable is always set. 
I think the following will do the thing :
`group_by = kwargs.get('group_by', False)`
and 
`'group_by': group_by or 'all',`
and in addons/website_profile/views/website_profile.xml : 
`<button type="button" t-attf-class="btn btn-secondary #{ 'active' if group_by == 'all' else ''}" onclick="location.href='/profile/users'">All time</button>`
and 
`<t t-else="'">All time </t>` (as setting a condition in a t-else is useless anyway)
So all will be always active, unless we ask to group by week or month.